src/generators/NextGenerator.js

```diff
@@ -0,0 +1,232 @@
+        console.log(
+          chalk.yellow(
+            "It seems express is not installed but generator needs a custom express server to work efficiently."
+          )
+        );
+      }
+    }
+  }
+
+  checkImports(directory, imports, extension = ".ts") {
+    imports.forEach(({ file }) => {
+      if (!fs.existsSync(directory + file + extension)) {
+        console.log(
+          chalk.yellow(
+            'An import for the file  "%s" has been generated but the file doesn\'t exists.'
+          ),
+          file
+        );
+      }
+    });
+  }
+
+  help(resource, dir) {
+    console.log(
+      chalk.green('Code for the "%s" resource type has been generated!'),
+      resource.title
+    );
+
+    // missing import
+    const { imports } = this.parseFields(resource);
+    this.checkImports(`${dir}/interfaces/`, imports);
+
+    // server route configuration
+    const lc = resource.title.toLowerCase();
+    console.log("Paste the following route to your server configuration file:");
+    console.log(
+      chalk.green(`
+server.get('/${lc}/:hash', (req, res) => {
+  return app.render(req, res, '/${lc}', { hash: req.params.hash })
+});
+`)
+    );
+  }
+
+  generate(api, resource, dir) {
+    const lc = resource.title.toLowerCase();
+    const ucf = this.ucFirst(resource.title);
+    const { fields, imports } = this.parseFields(resource);
+
+    const context = {
+      name: resource.name,
+      lc,
+      uc: resource.title.toUpperCase(),
+      ucf,
+      fields,
+      formFields: this.buildFields(resource.writableFields),
+      imports,
+      hydraPrefix: this.hydraPrefix,
+      title: resource.title
+    };
+
+    // Create directories
+    // These directories may already exist
+    [
+      `${dir}/components/common`,
+      `${dir}/config`,
+      `${dir}/error`,
+      `${dir}/interfaces`,
+      `${dir}/pages`,
+      `${dir}/utils`
+    ].forEach(dir => this.createDir(dir, false));
+
+    // copy with patterned name
+    this.createDir(`${dir}/components/${context.lc}`);
+    [
+      // components
+      "components/%s/List.tsx",
+      "components/%s/ListItem.tsx",
+      "components/%s/Show.tsx",
+
+      // pages
+      "pages/%s.tsx",
+      "pages/%ss.tsx"
+    ].forEach(pattern =>
+      this.createFileFromPattern(pattern, dir, context.lc, context)
+    );
+
+    // interface pattern should be camel cased
+    this.createFile(
+      "interfaces/foo.ts",
+      `${dir}/interfaces/${context.ucf}.ts`,
+      context
+    );
+
+    // copy with regular name
+    [
+      // components
+      "components/common/ReferenceLinks.tsx",
+
+      // error
+      "error/SubmissionError.ts",
+
+      // interfaces
+      "interfaces/Collection.ts",
+
+      // utils
+      "utils/dataAccess.ts"
+    ].forEach(file => this.createFile(file, `${dir}/${file}`, context, false));
+
+    // API config
+    this.createEntrypoint(api.entrypoint, `${dir}/config/entrypoint.ts`);
+  }
+
+  getType(field) {
```

Comment:
At some point, we should probably move this utility in a function reusable by all generators.